### PR TITLE
feat: replace Travis with Github actions

### LINKED ISSUE #1270: Switch from Travis to GitHub Actions
We recently created [setup-ocaml](https://github.com/avsm/setup-ocaml). If you are okay, I would like to switch to that.

### PR DESCRIPTION
**Pull request by Naturalclar**
Closes #1270 

Attempt to switch from Travis to Github Actions.

Please see [CONTRIBUTING.md](./CONTRIBUTING.md) for how to set up your development environment and run tests.

### CODE REVIEW

File: .github/workflows/install.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Use OCaml ${{ matrix.ocaml-version }}
+        uses: avsm/setup-ocaml@v1
+        with:
+          ocaml-version: ${{ matrix.ocaml-version }}
+
+      - name: Restore opam cache
+        uses: actions/cache@v2
+        with:
+          path: ${HOME}/.opam
+          key: ${{ runner.os }}-${{ hashFiles('.opam') }}
```
Comment (imbsky) [inline .github/workflows/install.yml:39]:
```suggestion
          key: ${{ runner.os }}-${{ hashFiles('opam.locked') }}
```

File: .github/workflows/install.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - run: opam pin add --no-action infer .
+
+      - run: opam depext --update infer;
+        if: runner.os != 'macOS'
+
+      - run: opam install --deps-only infer
+
+      - run: opam install infer
+
+      - run: opam remove infer || true
```
Comment (imbsky) [inline .github/workflows/install.yml:58]: We don't need this line.

File: .github/workflows/install.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        uses: actions/cache@v2
+        with:
+          path: ${HOME}/.opam
+          key: ${{ runner.os }}-${{ hashFiles('opam.locked') }}
+
+      - name: Install Required Brew Packages for MacOS
+        run: brew install pkg-config automake
+        if: runner.os == 'macOS'
+
+      - run: opam remove infer || true
```
Comment (imbsky) [inline .github/workflows/install.yml:45]:
```suggestion

      # ensure infer isn't installed in this switch, then deal with dependencies
      - run: opam remove infer || true
```

File: .github/workflows/install.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+on:
+  - push
+  - pull_request
+
+jobs:
+  build:
+    env:
+      OCAML_VERSION: 4.09.1
+      OPAMYES: 1
+      OPAMJOBS: 2
```
Comment (imbsky) [inline .github/workflows/install.yml:12]: These environment variables are not actually used and/or are unnecessary.

File: .github/workflows/install.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          - ubuntu-latest
+        ocaml-version:
+          - 4.09.1
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
```
Comment (imbsky) [inline .github/workflows/install.yml:29]:
```suggestion
      - name: Checkout code
        uses: actions/checkout@v2
        with:
          submodules: true

```

File: .github/workflows/install.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        if: runner.os == 'macOS'
+
+      - run: opam remove infer || true
+
+      - run: opam update --upgrade
+
+      - run: opam pin add --no-action infer .
+
+      - run: opam depext --update infer;
+        if: runner.os != 'macOS'
```
Comment (imbsky) [inline .github/workflows/install.yml:52]:
```suggestion
```

File: .github/workflows/install.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  - pull_request
+
+jobs:
+  build:
+    env:
+      OCAML_VERSION: 4.09.1
+      OPAMYES: 1
+      OPAMJOBS: 2
+      OPAMVERBOSE: 1
+      INFER_OPAM_SWITCH: "ocaml-variants.${OCAML_VERSION}+flambda"
```
Comment (Naturalclar) [inline .github/workflows/install.yml:14]:
```suggestion
    env:
      OCAML_VERSION: 4.09.1
      OPAMYES: 1
      OPAMJOBS: 2
      OPAMVERBOSE: 1
      INFER_OPAM_SWITCH: "ocaml-variants.${OCAML_VERSION}+flambda"
```

File: .github/workflows/install.yml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      - run: opam pin add --no-action infer .
+
+      - run: opam depext --update infer;
+        if: runner.os != 'macOS'
+
+      - run: opam install --deps-only infer
+
+      - run: opam install infer
+
+      - run: opam remove infer || true
```
Comment (Naturalclar) [inline .github/workflows/install.yml:58]:
```suggestion
```

**Comment by XVilka** [comment]
Maybe run tests also?

**Comment by imbsky** [comment]
The current Travis setup is not running any tests. The purpose of this PR is just a replacement, so it's out of scope. We can do it with another PR.

**Comment by Naturalclar** [comment]
I agree with @imbsky . It'd be nice to have tests and other checks to be performed with GHA, but it's out of the scope for this PR.

**Comment by jvillard** [comment]
This is great, thanks :clap:

Looks ready to merge, right?

**Comment by imbsky** [comment]
Probably yes.

**Pull request closed without merging**